test/codeu/chat/server/JSONTest.java

```diff
@@ -0,0 +1,204 @@
+ */
+public class JSONTest {
+
+    @Test
+    public void readTest1() {
+        // This function tests if the JSON reader is working if there is a JSON with one type of each Object in the file
+        Model model = new Model();
+        JSON log = new JSON();
+        Model comparer = new Model();
+        comparer.add(new User(createUuid("1.1074501217"), "Krager", Time.fromMs(4309509)));
+        comparer.add(new ConversationHeader(createUuid("1.3959833157"), createUuid("1.1074501217"),  Time.fromMs(59408509), "Chat1"));
+        comparer.add(new Message(createUuid("1.3928689216"), Uuid.NULL, Uuid.NULL, Time.fromMs(49584908), createUuid("1.1074501217"),   "I'm so alone"));
+
+        try {
+            // Path used by Priyanka:
+            model = log.readFromFile("/Users/HMCLoaner/Desktop/CodeU/test/codeu/chat/server/input.json");
+            // The path can vary from device, so this may not compile if it's not the correct path for your project
+            // set up. A suggested path is as follows
+            // model = log.readFromFile("CodeU/test/codeu/chat/server/input.json");
+        } catch (IOException e) {
+            System.out.println("File not handled properly");
+        }
+        assertEquals(
+                compare(model.userById().all().iterator(), comparer.userById().all().iterator()), true);
+        assertEquals(
+                compare(model.userByText().all().iterator(), comparer.userByText().all().iterator()), true);
+        assertEquals(
+                compare(model.userByTime().all().iterator(), comparer.userByTime().all().iterator()), true);
+        assertEquals(
+                compare(model.conversationById().all().iterator(), comparer.conversationById().all().iterator()), true);
+        assertEquals(
+                compare(model.conversationByTime().all().iterator(), comparer.conversationByTime().all().iterator()), true);
+        assertEquals(
+                compare(model.conversationByText().all().iterator(), comparer.conversationByText().all().iterator()), true);
+        assertEquals(
+                compare(model.messageById().all().iterator(), comparer.messageById().all().iterator()), true);
+        assertEquals(
+                compare(model.messageByText().all().iterator(), comparer.messageByText().all().iterator()), true);
+        assertEquals(
+                compare(model.messageByTime().all().iterator(), comparer.messageByTime().all().iterator()), true);
+    }
+
+    @Test
+    public void readTest2() {
+        // This function tests if the JSON reader is working if there is a JSON with one type of User in the file
+        Model model = new Model();
+        JSON log = new JSON();
+        Model comparer = new Model();
+        comparer.add(new User(createUuid("1.1074501217"), "Krager", Time.fromMs(12345)));
+        try {
+            // Path used by Priyanka:
+            model = log.readFromFile("/Users/HMCLoaner/Desktop/CodeU/test/codeu/chat/server/input2.json");
```

Comment:
I think relative paths are relative to the directory the code was run from, not the location of the class file.  You might also have to modify the `build.py` file to copy over `.json` files into the `bin` directory. Because when the code is executing, that's where it's executing from. Basically it has to be relative to the `.class` file, not the `.java` file.